docs/book/src/developer/providers/v0.4-to-v1.0.md

```diff
@@ -57,7 +57,8 @@ The `serving-cert` certificates now have organization set to `k8s-sig-cluster-li
   [machine][machine-contract] provider contract docs for more information. This is not required, but is recommended for
   consistency across the infrastructure providers as Cluster API graduates and opens up use cases where coordinating
   controllers can use labels and annotations from template infrastructure resources to do external provisioning or
-  provide configuration information, e.g. [IPAM support for vSphere / bare-metal][capv-ipam].
+  provide configuration information, e.g. [IPAM support for vSphere / bare-metal][capv-ipam]. 
+- Labels and annotations from KubeadmControlPlane, MachineDeployment and MachineSet and their `.spec.template.metadata` fields are now selectively propagated to objects controlled by their respective controllers. Refer to [metadata-propagation] for more information.
```

Comment:
If I understand correctly some of the behaviour (but not all) changed in https://github.com/kubernetes-sigs/cluster-api/issues/5240. I can be more clear or specific on that, but I thought it might be better just to link to the overall documentation for review WDYT?